buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -0,0 +1,286 @@
+        BuildPlugin.configureCompile(project);
+        BuildPlugin.configureRepositories(project);
+        BuildPlugin.configureTestTasks(project);
+        BuildPlugin.configureInputNormalization(project);
+
+        TaskProvider<Test> destructiveTest = project.getTasks().register("destructiveDistroTest", Test.class);
+        destructiveTest.configure(t -> {
+            t.setMaxParallelForks(1);
+            if (System.getProperty(IN_VM_SYSPROP) == null) {
+                t.dependsOn(COPY_PACKAGING_TASK);
+            }
+        });
+
+        // setup outer task to run
+        TaskProvider<GradleDistroTestTask> testCaller = project.getTasks().register("distroTest", GradleDistroTestTask.class);
+        testCaller.configure(t -> {
+            t.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+            t.setDescription("Runs distribution tests within vagrant");
+            t.setTaskName(project.getPath() + ":" + destructiveTest.getName());
+            t.extraArg("-D" + IN_VM_SYSPROP);
+            t.dependsOn(COPY_PACKAGING_TASK);
+        });
+    }
+
+    private static void configureBatsTest(Project project, String type) {
+
+        // destructive task to run inside
+        TaskProvider<BatsTestTask> destructiveTest = project.getTasks().register("destructiveBatsTest." + type, BatsTestTask.class);
+        destructiveTest.configure(t -> {
+            // this is hacky for shared source, but bats are a temporary thing we are removing, so it is not worth
+            // the overhead of a real project dependency
+            Directory batsDir = project.getParent().getLayout().getProjectDirectory().dir("bats");
+            t.setTestsDir(batsDir.dir(type));
+            t.setUtilsDir(batsDir.dir("utils"));
+            t.setArchivesDir(project.getLayout().getBuildDirectory().dir("packaging/archives").get());
+            t.setPackageName("elasticsearch" + (type.equals("oss") ? "-oss" : ""));
+            if (System.getProperty(IN_VM_SYSPROP) == null) {
+                t.dependsOn(COPY_PACKAGING_TASK);
+            }
+        });
+
+        // setup outer task to run
+        TaskProvider<GradleDistroTestTask> testCaller = project.getTasks().register("batsTest." + type, GradleDistroTestTask.class);
+        testCaller.configure(t -> {
+            t.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+            t.setDescription("Runs bats tests within vagrant");
+            t.setTaskName(project.getPath() + ":" + destructiveTest.getName());
+            t.setProgressHandler(new BatsProgressLogger(project.getLogger()));
+            t.extraArg("-D" + IN_VM_SYSPROP);
+            t.dependsOn(COPY_PACKAGING_TASK);
+        });
+    }
+    
+    private void configureDistributions(Project project) {
+        NamedDomainObjectContainer<ElasticsearchDistribution> distributions = DistributionDownloadPlugin.getContainer(project);
+
+        for (Type type : Arrays.asList(Type.DEB, Type.RPM)) {
```

Comment:
I think it would be better to encapsulate this in `Type` by having a method that returns the `Arrays,asList(...)` from here

Reply:
If we were using this often I might agree, but I don't think it is necessary here. Adding such a method would mean coming up with a name to describe them, but I don't really need that, I just know that here I want both of these distro types.

Reply:
I was thinking that the information belongs more to how we model the type than the test we run, but agree on the naming and don't have a strong preference.